src/lib.rs

```diff
@@ -390,6 +507,62 @@ mod tests {
         SMFunc { addr, stack_size }
     }
 
+    fn parse_loc(line: &str) -> SMLoc {
+        let elems: Vec<&str> = line.split(|c| c == ',' || c == ':').collect();
+        // Gives ["#N " LocKind +Data", "0", "size", " 8"]
+
+        let size = elems[3].trim().parse::<u16>().unwrap();
+        let mut loc = elems[1].trim().split_whitespace();
+        let kind = loc.next().unwrap();
+        let rest: Vec<&str> = loc.collect();
```

Comment:
You can probably do `Vec<_>` to avoid writing out the whole type. [It's probably more idiomatic to do `collect::<Vec<_>>()` but that's partly a matter of taste.]